compiler/rustc_graphviz/src/lib.rs

```diff
@@ -164,7 +164,7 @@
 //!     fn node_id(&'a self, n: &Nd) -> dot::Id<'a> {
 //!         dot::Id::new(format!("N{}", n)).unwrap()
 //!     }
-//!     fn node_label<'b>(&'b self, n: &Nd) -> dot::LabelText<'b> {
+//!     fn node_label(&self, n: &Nd) -> dot::LabelText {
```

Comment:
I think this is less clear after your change; implicit lifetimes are hard to reason about.

Reply:
It's also not the same thing. If the caller uses a short lived `n`, the return type is needlessly constrained to that lifetime.

Reply:
I thought elided lifetime rules preferred &self over other input lifetimes?

Reply:
Oh ^^ ignore me then

Reply:
So I'll admit I wasn't familiar with the nuances of the lifetime elision rules before reading this PR, but looking over https://doc.rust-lang.org/nomicon/lifetime-elision.html they seem to make sense. I guess the question is, do we expect compiler developers to know that return values share the same lifetime as `self` unless otherwise written? This seems not unreasonable to me, even though I didn't know it until now.

Reply:
Oh no, I literally meant that removing `<'b>` hides that it has an inferred lifetime; I would be ok with `LabelText<'_>`.

Reply:
We should probably just enable https://doc.rust-lang.org/beta/nightly-rustc/rustc_lint/builtin/static.ELIDED_LIFETIMES_IN_PATHS.html globally in rustc, I would be a fan of that change to make sure we enforce `LabelText<'_>` over `LabelText`

Reply:
https://github.com/rust-lang/rust/blob/f94c926aecf226a12ffc5c7f09156390dde545ca/src/bootstrap/builder.rs#L1662-L1667 would be the place to add that.

Reply:
> Oh no, I literally meant that removing `<'b>` hides that it has an inferred lifetime; I would be ok with `LabelText<'_>`.

Oh yeah, that makes sense. I agree, I'd prefer the `'_` were visible here.